python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py

```diff
@@ -475,20 +475,27 @@ def _train_batch(self, batch, batch_info=None, callbacks=None):
                 calculate averages.
 
         """
-        # TODO: restore batch to what it should be.
-        self.call_hook(call_backs=callbacks, fn_name="before_train_iter")
         # unpack features into list to support multiple inputs model
+        # and restore batch to what it should be.
         features, target = batch
```

Comment:
is this line correct? Why you first split and then restore and then split (in Line493)?

Reply:
> is this line correct? Why you first split and then restore and then split (in Line493)?

`first split and restore` is meant to recover batch to how user defined in dataloader
the last split(with a star-pack * operation) is to ensure features after `before_train_iter` are fed into self.model (No interal operations on features)

And all actions passed indicates that this adjustment works well as before I think.